### ISSUE TITLE
Request new module: Code Engine

### ISSUE THREAD

**Issue by markjmeredith** [cloud-native-toolkit/software-everywhere#569]
(Click Preview to navigate to guides) Please read the guides on how to create a module for the catalog:
- [Guide for a GitOps Module](https://modules.cloudnativetoolkit.dev/#/how-to/gitops) 
- [Guide for a Terraform Module](https://modules.cloudnativetoolkit.dev/#/how-to/terraform)

**Related user story**
A user sets up Code Engine to deploy instances of applications and wants to use terraform to create and mange deployments. https://cloud.ibm.com/apidocs/codeengine/codeengine-v2.0.0 

**Type of module**
terraform

**If requesting a terraform module, what is the target cloud platform?**
ibm

**What is the purpose of the module?**
CRUD applications and jobs in Code Engine (also CRUD on CE instance)

**What is the category of the module?**
cluster might be closest..

**Comment by timroster** [comment]
This would likely be split into two modules, one to create a code engine project(s) and another to manage application lifecycle for deployments to the project.

**Comment by triceam** [comment]
Agreed.  @markjmeredith it might be worthwhile to discuss requirements on this one.
